File: src/main/java/com/exception_study/global/exception/ErrorCode.java
package com.exception_study.global.exception;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;

@Getter
@RequiredArgsConstructor
public enum ErrorCode {
    INVALID_TOKEN(HttpStatus.UNAUTHORIZED, "Invalid token"),
    USER_NOT_FOUND(HttpStatus.NOT_FOUND, "User not founded"),
    USER_ALREADY_EXIST(HttpStatus.CONFLICT,"User already exists"),
    PRODUCT_NOT_FOUND(HttpStatus.NOT_FOUND,"Product not founded"),
    PRODUCT_SOLD_OUT(HttpStatus.CONFLICT,"Product sold out"),
    PRODUCT_ALREADY_RESERVED(HttpStatus.CONFLICT,"Already reserved product"),
    PRODUCT_ORDER_ALREADY_CONFIRMED(HttpStatus.CONFLICT,"Already confirmed order"),
    ORDER_NOT_FOUND(HttpStatus.NOT_FOUND,"ORDER not founded"),
    INVALID_PASSWORD(HttpStatus.UNAUTHORIZED, "Invalid password"),
    INVALID_PERMISSION(HttpStatus.UNAUTHORIZED, "User has invalid permission"),
    DATABASE_ERROR(HttpStatus.INTERNAL_SERVER_ERROR, "Database error occurs");

    private final HttpStatus status;
    private final String message;
}

File: src/main/kotlin/wanted/market/common/exception/ErrorCode.kt
package wanted.market.common.exception

enum class ErrorCode(val httpStatus: Int, val message: String) {

    OK(200, "Operation successful"),

    MEMBER_NOT_FOUND(404, "존재하지 않는 회원입니다."),
    DUPLICATE_EMAIL(500, "중복되는 이메일입니다."),

    PRODUCT_NOT_FOUND(404, "존재하지 않는 상품입니다."),
    PRODUCT_NOT_FOR_SALE(400, "판매 중인 상품이 아닙니다."),
    SELLER_APPROVAL_ONLY(400, "판매자만 승인할 수 있습니다."),


    ORDER_NOT_FOUND(404, "존재하지 않는 주문입니다."),

    BAD_REQUEST(400, "클라리언트의 잘못된 요청입니다."),
    INTERNAL_SERVER_ERROR(500, "서버 내부 에러입니다.")
}